### ISSUE TITLE
Add a theme props to root chart types

### ISSUE THREAD

**Issue by mcnuttandrew** [uber/react-vis#242]
The uber branded color type is really cool, but what if users want material design? or a color blind friendly theme? I propose we expose a prop on all top level charts to modify our concept of theme. The api would be like 

```
<XYPlot theme="material">
...
</XYPlot>
```

**Comment by Apercu** [comment]
👍  and a top level component that exposes those properties to all child components

**Comment by mcnuttandrew** [comment]
So this would be something like 

```
<ReactVis theme='matierial'>
<XYPlot> ... </XYPlot>
<XYPlot> ... </XYPlot>
</ReactVis>

```
question mark?

**Comment by mcnuttandrew** [comment]
Let's get a list of themes going? 
- Material
- Data Vis
- B + W
Anything else?

**Comment by Apercu** [comment]
Yes, and these are the ones we can start with yes.

**Comment by akre54** [comment]
Maybe allow passing a theme object (merged with defaults from theme.js), or pull out more scss variables instead of passing a theme string?

**Comment by Apercu** [comment]
Yep, we were planning on that too, theme strings only represent the default ones, thanks for the reminder!

**Comment by mcnuttandrew** [comment]
Added custom to the list!

**Comment by mcnuttandrew** [comment]
Given that we are starting to treat XYPlot as the root for chart types, it seems like this could effectively handled there!

**Comment by leetercola** [comment]
Can I ask where this feature landed. From the documentation there is no clear way to providing a default theme or color palette to the components as a whole. I see this enhancement ticket https://github.com/uber/react-vis/issues/823 which doesn't seem to be resolved but a user mentions creating https://github.com/swiftcarrot/react-vis-styles to solve the use case. Can we get some documentation around official direction of theming or global overrides of color palettes
